src/llama.cpp

```diff
@@ -15581,6 +15842,16 @@ struct llm_tokenizer_bpe {
                     "\\p{N}",
                 };
                 break;
+            case LLAMA_VOCAB_PRE_TYPE_CHAMELEON:
+                regex_exprs = {
+                    "<sentinel:[0-9]+>",  // Sentinel tokens
+                    "(IMGIMG)((A|B|C|D|E|F|G|H|I){1,4})Z",  // Image tokens
```

Comment:
Side note: special tokens don't usually need to be considered by the split regex (since HF `tokenizers` pre-tokenize them specially anyway, and `llama.cpp` too in `tokenizer_st_partition`), but it seems like the upstream tokenizer uses these regular expressions in the `pre_tokenzier`, so they likely did not know this. Keeping this is appropriate, especially since their sentinel token regex matches more than only the sentinel tokens in the vocab.

Other note: this is the model with *the biggest number* of special tokens I have ever seen. 16384 special tokens!?

Reply:
>special tokens don't usually need to be considered by the split regex (since HF tokenizers pre-tokenize them specially anyway, and llama.cpp too in tokenizer_st_partition)

Good to know! Unsure if they are even intentionally used, but probably best to keep them. I'll add a comment.

>Other note: this is the model with the biggest number of special tokens I have ever seen. 16384 special tokens!? That's like 1/4 of its vocab!
true  😄  A big part of them are the image tokens from the VQVAE codebook (8192 tokens).